### ISSUE TITLE
Add ignoreAtRules: [] option to at-rule-empty-line-before

### ISSUE THREAD

**Issue by dikmax** [stylelint/stylelint#1377]
A new options for `at-rule-empty-line-before` rule is needed. Like `ignoreAtRules` in `block-closing-brace-newline-after` and `block-opening-brace-space-before`. (See #561)

it will allow to lint scss `@if...@else` like:

```css
@if true {
  display: block;
} @else {
  display: none;
}
```

Config used:

```json
{
  "extends": "stylelint-config-standard"
}
```

Stylelint version: 6.5.1

Run with CLI: `stylelint ./**/*.scss --syntax scss`

Warnings:

```shell
test.scss
 3:3  ✖  Expected empty line before at-rule   at-rule-empty-line-before
```

Expected result:

Configuration option `ignoreAtRules` for the rule to prevent only this kind of warnings, but not all.

**Comment by evilebottnawi** [comment]
@dikmax please format you issue as template

**Comment by davidtheclark** [comment]
Also please search the issues. This has come up before.

**Comment by dikmax** [comment]
@davidtheclark I've searched prior creating an issue. None was found.

**Comment by davidtheclark** [comment]
See #798 and #561. Might help give concrete ideas for what options to add.

**Comment by evilebottnawi** [comment]
@dikmax Feel free to PR

**Comment by jeddy3** [comment]
@dikmax This is very much a legitimate request and I find it kind of odd that it has not come up before. I guess no one else is using the `at-rule-empty-line-before` rule alongside the `ignoreAtRules: []` option for the `block-*` rules.

Anyhow, as @evilebottnawi said, feel free to contribute this feature. [This PR](https://github.com/stylelint/stylelint/pull/803/files) should give you everything you need to know. Feel free to use `ignoreAtRules: ["else"]` in your documentation examples, as I think that'll be the most common use case here. e.g.

```css
@if true {
  display: block;
} @else {
  display: none;
}

@if true {
  display: block;
} @else {
  display: none;
}
```

**Comment by Nirazul** [comment]
I also though about this some time ago. To me, it seems that also an option like `ignore: ["blockless-group"]` is missing. 
I'd like to enforce an empty line for at-rules with blocks but I don't care for, say lists of `@import` statements.

**Comment by jeddy3** [comment]
@Nirazul I think your request is more related to https://github.com/stylelint/stylelint/issues/1366. Either way, it's a valid request and complimentary to https://github.com/stylelint/stylelint/issues/1366. I've created [a new issue](https://github.com/stylelint/stylelint/issues/1413) for it. Please feel free to contribute this option.

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/issues/1423 by @Nirazul.
